plugins/inputs/riak/riak.go

```diff
@@ -36,52 +36,105 @@ func NewRiak() *Riak {
 
 // Type riakStats represents the data that is received from Riak
 type riakStats struct {
-	CPUAvg1                  int64  `json:"cpu_avg1"`
-	CPUAvg15                 int64  `json:"cpu_avg15"`
-	CPUAvg5                  int64  `json:"cpu_avg5"`
-	MemoryCode               int64  `json:"memory_code"`
-	MemoryEts                int64  `json:"memory_ets"`
-	MemoryProcesses          int64  `json:"memory_processes"`
-	MemorySystem             int64  `json:"memory_system"`
-	MemoryTotal              int64  `json:"memory_total"`
-	NodeGetFsmObjsize100     int64  `json:"node_get_fsm_objsize_100"`
-	NodeGetFsmObjsize95      int64  `json:"node_get_fsm_objsize_95"`
-	NodeGetFsmObjsize99      int64  `json:"node_get_fsm_objsize_99"`
-	NodeGetFsmObjsizeMean    int64  `json:"node_get_fsm_objsize_mean"`
-	NodeGetFsmObjsizeMedian  int64  `json:"node_get_fsm_objsize_median"`
-	NodeGetFsmSiblings100    int64  `json:"node_get_fsm_siblings_100"`
-	NodeGetFsmSiblings95     int64  `json:"node_get_fsm_siblings_95"`
-	NodeGetFsmSiblings99     int64  `json:"node_get_fsm_siblings_99"`
-	NodeGetFsmSiblingsMean   int64  `json:"node_get_fsm_siblings_mean"`
-	NodeGetFsmSiblingsMedian int64  `json:"node_get_fsm_siblings_median"`
-	NodeGetFsmTime100        int64  `json:"node_get_fsm_time_100"`
-	NodeGetFsmTime95         int64  `json:"node_get_fsm_time_95"`
-	NodeGetFsmTime99         int64  `json:"node_get_fsm_time_99"`
-	NodeGetFsmTimeMean       int64  `json:"node_get_fsm_time_mean"`
-	NodeGetFsmTimeMedian     int64  `json:"node_get_fsm_time_median"`
-	NodeGets                 int64  `json:"node_gets"`
-	NodeGetsTotal            int64  `json:"node_gets_total"`
-	Nodename                 string `json:"nodename"`
-	NodePutFsmTime100        int64  `json:"node_put_fsm_time_100"`
-	NodePutFsmTime95         int64  `json:"node_put_fsm_time_95"`
-	NodePutFsmTime99         int64  `json:"node_put_fsm_time_99"`
-	NodePutFsmTimeMean       int64  `json:"node_put_fsm_time_mean"`
-	NodePutFsmTimeMedian     int64  `json:"node_put_fsm_time_median"`
-	NodePuts                 int64  `json:"node_puts"`
-	NodePutsTotal            int64  `json:"node_puts_total"`
-	PbcActive                int64  `json:"pbc_active"`
-	PbcConnects              int64  `json:"pbc_connects"`
-	PbcConnectsTotal         int64  `json:"pbc_connects_total"`
-	VnodeGets                int64  `json:"vnode_gets"`
-	VnodeGetsTotal           int64  `json:"vnode_gets_total"`
-	VnodeIndexReads          int64  `json:"vnode_index_reads"`
-	VnodeIndexReadsTotal     int64  `json:"vnode_index_reads_total"`
-	VnodeIndexWrites         int64  `json:"vnode_index_writes"`
-	VnodeIndexWritesTotal    int64  `json:"vnode_index_writes_total"`
-	VnodePuts                int64  `json:"vnode_puts"`
-	VnodePutsTotal           int64  `json:"vnode_puts_total"`
-	ReadRepairs              int64  `json:"read_repairs"`
-	ReadRepairsTotal         int64  `json:"read_repairs_total"`
+	ClusterAaeFsmActive        int64  `json:"clusteraae_fsm_active"`
```

Comment:
Could you include a link to the riak documentation in a comment here? Then when people edit this in the future they will know where you got the stats from.

Reply:
Sadly, Riak's documentation on this aspect is dated and doesn't include the full lists of available metrics (see [here](https://www.tiot.jp/riak-docs/riak/kv/3.0.4/using/reference/statistics-monitoring/)). I got the full list of metrics by running `riak admin status` in our Riak `3.0.10` cluster. I could paste the output here in this `PR` and perhaps mention this in the README.

Reply:
This is the list of `502` metrics currently available in Riak 3.x (`riak admin status` output).

```
clusteraae_fsm_active
clusteraae_fsm_create
clusteraae_fsm_create_error
connected_nodes
consistent_get_objsize_100
consistent_get_objsize_95
consistent_get_objsize_99
consistent_get_objsize_mean
consistent_get_objsize_median
consistent_get_time_100
consistent_get_time_95
consistent_get_time_99
consistent_get_time_mean
consistent_get_time_median
consistent_gets
consistent_gets_total
consistent_put_objsize_100
consistent_put_objsize_95
consistent_put_objsize_99
consistent_put_objsize_mean
consistent_put_objsize_median
consistent_put_time_100
consistent_put_time_95
consistent_put_time_99
consistent_put_time_mean
consistent_put_time_median
consistent_puts
consistent_puts_total
converge_delay_last
converge_delay_max
converge_delay_mean
converge_delay_min
coord_local_soft_loaded_total
coord_local_unloaded_total
coord_redir_least_loaded_total
coord_redir_loaded_local_total
coord_redir_unloaded_total
coord_redirs_total
counter_actor_counts_100
counter_actor_counts_95
counter_actor_counts_99
counter_actor_counts_mean
counter_actor_counts_median
cpu_avg1
cpu_avg15
cpu_avg5
cpu_nprocs
dropped_vnode_requests_total
executing_mappers
gossip_received
handoff_timeouts
hll_bytes
hll_bytes_100
hll_bytes_95
hll_bytes_99
hll_bytes_mean
hll_bytes_median
hll_bytes_total
ignored_gossip_total
index_fsm_active
index_fsm_complete
index_fsm_create
index_fsm_create_error
index_fsm_results_100
index_fsm_results_95
index_fsm_results_99
index_fsm_results_mean
index_fsm_results_median
index_fsm_time_100
index_fsm_time_95
index_fsm_time_99
index_fsm_time_mean
index_fsm_time_median
late_put_fsm_coordinator_ack
leveldb_read_block_error
list_fsm_active
list_fsm_create
list_fsm_create_error
list_fsm_create_error_total
list_fsm_create_total
map_actor_counts_100
map_actor_counts_95
map_actor_counts_99
map_actor_counts_mean
map_actor_counts_median
mem_allocated
mem_total
memory_atom
memory_atom_used
memory_binary
memory_code
memory_ets
memory_processes
memory_processes_used
memory_system
memory_total
ngrfetch_nofetch
ngrfetch_nofetch_total
ngrfetch_prefetch
ngrfetch_prefetch_total
ngrfetch_tofetch
ngrfetch_tofetch_total
ngrrepl_empty
ngrrepl_empty_total
ngrrepl_error
ngrrepl_error_total
ngrrepl_object
ngrrepl_object_total
ngrrepl_srcdiscard
ngrrepl_srcdiscard_total
node_get_fsm_active
node_get_fsm_active_60s
node_get_fsm_counter_objsize_100
node_get_fsm_counter_objsize_95
node_get_fsm_counter_objsize_99
node_get_fsm_counter_objsize_mean
node_get_fsm_counter_objsize_median
node_get_fsm_counter_siblings_100
node_get_fsm_counter_siblings_95
node_get_fsm_counter_siblings_99
node_get_fsm_counter_siblings_mean
node_get_fsm_counter_siblings_median
node_get_fsm_counter_time_100
node_get_fsm_counter_time_95
node_get_fsm_counter_time_99
node_get_fsm_counter_time_mean
node_get_fsm_counter_time_median
node_get_fsm_errors
node_get_fsm_errors_total
node_get_fsm_hll_objsize_100
node_get_fsm_hll_objsize_95
node_get_fsm_hll_objsize_99
node_get_fsm_hll_objsize_mean
node_get_fsm_hll_objsize_median
node_get_fsm_hll_siblings_100
node_get_fsm_hll_siblings_95
node_get_fsm_hll_siblings_99
node_get_fsm_hll_siblings_mean
node_get_fsm_hll_siblings_median
node_get_fsm_hll_time_100
node_get_fsm_hll_time_95
node_get_fsm_hll_time_99
node_get_fsm_hll_time_mean
node_get_fsm_hll_time_median
node_get_fsm_in_rate
node_get_fsm_map_objsize_100
node_get_fsm_map_objsize_95
node_get_fsm_map_objsize_99
node_get_fsm_map_objsize_mean
node_get_fsm_map_objsize_median
node_get_fsm_map_siblings_100
node_get_fsm_map_siblings_95
node_get_fsm_map_siblings_99
node_get_fsm_map_siblings_mean
node_get_fsm_map_siblings_median
node_get_fsm_map_time_100
node_get_fsm_map_time_95
node_get_fsm_map_time_99
node_get_fsm_map_time_mean
node_get_fsm_map_time_median
node_get_fsm_objsize_100
node_get_fsm_objsize_95
node_get_fsm_objsize_99
node_get_fsm_objsize_mean
node_get_fsm_objsize_median
node_get_fsm_out_rate
node_get_fsm_rejected
node_get_fsm_rejected_60s
node_get_fsm_rejected_total
node_get_fsm_set_objsize_100
node_get_fsm_set_objsize_95
node_get_fsm_set_objsize_99
node_get_fsm_set_objsize_mean
node_get_fsm_set_objsize_median
node_get_fsm_set_siblings_100
node_get_fsm_set_siblings_95
node_get_fsm_set_siblings_99
node_get_fsm_set_siblings_mean
node_get_fsm_set_siblings_median
node_get_fsm_set_time_100
node_get_fsm_set_time_95
node_get_fsm_set_time_99
node_get_fsm_set_time_mean
node_get_fsm_set_time_median
node_get_fsm_siblings_100
node_get_fsm_siblings_95
node_get_fsm_siblings_99
node_get_fsm_siblings_mean
node_get_fsm_siblings_median
node_get_fsm_time_100
node_get_fsm_time_95
node_get_fsm_time_99
node_get_fsm_time_mean
node_get_fsm_time_median
node_gets
node_gets_counter
node_gets_counter_total
node_gets_hll
node_gets_hll_total
node_gets_map
node_gets_map_total
node_gets_set
node_gets_set_total
node_gets_total
node_put_fsm_active
node_put_fsm_active_60s
node_put_fsm_counter_time_100
node_put_fsm_counter_time_95
node_put_fsm_counter_time_99
node_put_fsm_counter_time_mean
node_put_fsm_counter_time_median
node_put_fsm_hll_time_100
node_put_fsm_hll_time_95
node_put_fsm_hll_time_99
node_put_fsm_hll_time_mean
node_put_fsm_hll_time_median
node_put_fsm_in_rate
node_put_fsm_map_time_100
node_put_fsm_map_time_95
node_put_fsm_map_time_99
node_put_fsm_map_time_mean
node_put_fsm_map_time_median
node_put_fsm_out_rate
node_put_fsm_rejected
node_put_fsm_rejected_60s
node_put_fsm_rejected_total
node_put_fsm_set_time_100
node_put_fsm_set_time_95
node_put_fsm_set_time_99
node_put_fsm_set_time_mean
node_put_fsm_set_time_median
node_put_fsm_time_100
node_put_fsm_time_95
node_put_fsm_time_99
node_put_fsm_time_mean
node_put_fsm_time_median
node_puts
node_puts_counter
node_puts_counter_total
node_puts_hll
node_puts_hll_total
node_puts_map
node_puts_map_total
node_puts_set
node_puts_set_total
node_puts_total
nodename
object_counter_merge
object_counter_merge_time_100
object_counter_merge_time_95
object_counter_merge_time_99
object_counter_merge_time_mean
object_counter_merge_time_median
object_counter_merge_total
object_hll_merge
object_hll_merge_time_100
object_hll_merge_time_95
object_hll_merge_time_99
object_hll_merge_time_mean
object_hll_merge_time_median
object_hll_merge_total
object_map_merge
object_map_merge_time_100
object_map_merge_time_95
object_map_merge_time_99
object_map_merge_time_mean
object_map_merge_time_median
object_map_merge_total
object_merge
object_merge_time_100
object_merge_time_95
object_merge_time_99
object_merge_time_mean
object_merge_time_median
object_merge_total
object_set_merge
object_set_merge_time_100
object_set_merge_time_95
object_set_merge_time_99
object_set_merge_time_mean
object_set_merge_time_median
object_set_merge_total
pbc_active
pbc_connects
pbc_connects_total
pipeline_active
pipeline_create_count
pipeline_create_error_count
pipeline_create_error_one
pipeline_create_one
postcommit_fail
precommit_fail
read_repairs
read_repairs_counter
read_repairs_counter_total
read_repairs_fallback_notfound_count
read_repairs_fallback_notfound_one
read_repairs_fallback_outofdate_count
read_repairs_fallback_outofdate_one
read_repairs_hll
read_repairs_hll_total
read_repairs_map
read_repairs_map_total
read_repairs_primary_notfound_count
read_repairs_primary_notfound_one
read_repairs_primary_outofdate_count
read_repairs_primary_outofdate_one
read_repairs_set
read_repairs_set_total
read_repairs_total
rebalance_delay_last
rebalance_delay_max
rebalance_delay_mean
rebalance_delay_min
rejected_handoffs
riak_kv_vnodeq_max
riak_kv_vnodeq_mean
riak_kv_vnodeq_median
riak_kv_vnodeq_min
riak_kv_vnodeq_total
riak_kv_vnodes_running
riak_pipe_vnodeq_max
riak_pipe_vnodeq_mean
riak_pipe_vnodeq_median
riak_pipe_vnodeq_min
riak_pipe_vnodeq_total
riak_pipe_vnodes_running
ring_creation_size
ring_members
ring_num_partitions
ring_ownership
rings_reconciled
rings_reconciled_total
set_actor_counts_100
set_actor_counts_95
set_actor_counts_99
set_actor_counts_mean
set_actor_counts_median
skipped_read_repairs
skipped_read_repairs_total
soft_loaded_vnode_mbox_total
storage_backend
sys_driver_version
sys_global_heaps_size
sys_heap_type
sys_logical_processors
sys_monitor_count
sys_otp_release
sys_port_count
sys_process_count
sys_smp_support
sys_system_architecture
sys_system_version
sys_thread_pool_size
sys_threads_enabled
sys_wordsize
tictacaae_branch_compare
tictacaae_branch_compare_total
tictacaae_bucket
tictacaae_bucket_total
tictacaae_clock_compare
tictacaae_clock_compare_total
tictacaae_error
tictacaae_error_total
tictacaae_exchange
tictacaae_exchange_total
tictacaae_modtime
tictacaae_modtime_total
tictacaae_not_supported
tictacaae_not_supported_total
tictacaae_queue_microsec__max
tictacaae_queue_microsec_mean
tictacaae_root_compare
tictacaae_root_compare_total
tictacaae_timeout
tictacaae_timeout_total
ttaaefs_allcheck_total
ttaaefs_daycheck_total
ttaaefs_fail_time_100
ttaaefs_fail_total
ttaaefs_hourcheck_total
ttaaefs_nosync_time_100
ttaaefs_nosync_total
ttaaefs_rangecheck_total
ttaaefs_snk_ahead_total
ttaaefs_src_ahead_total
ttaaefs_sync_time_100
ttaaefs_sync_total
vnode_counter_update
vnode_counter_update_time_100
vnode_counter_update_time_95
vnode_counter_update_time_99
vnode_counter_update_time_mean
vnode_counter_update_time_median
vnode_counter_update_total
vnode_get_fsm_time_100
vnode_get_fsm_time_95
vnode_get_fsm_time_99
vnode_get_fsm_time_mean
vnode_get_fsm_time_median
vnode_gets
vnode_gets_total
vnode_head_fsm_time_100
vnode_head_fsm_time_95
vnode_head_fsm_time_99
vnode_head_fsm_time_mean
vnode_head_fsm_time_median
vnode_heads
vnode_heads_total
vnode_hll_update
vnode_hll_update_time_100
vnode_hll_update_time_95
vnode_hll_update_time_99
vnode_hll_update_time_mean
vnode_hll_update_time_median
vnode_hll_update_total
vnode_index_deletes
vnode_index_deletes_postings
vnode_index_deletes_postings_total
vnode_index_deletes_total
vnode_index_reads
vnode_index_reads_total
vnode_index_refreshes
vnode_index_refreshes_total
vnode_index_writes
vnode_index_writes_postings
vnode_index_writes_postings_total
vnode_index_writes_total
vnode_map_update
vnode_map_update_time_100
vnode_map_update_time_95
vnode_map_update_time_99
vnode_map_update_time_mean
vnode_map_update_time_median
vnode_map_update_total
vnode_mbox_check_timeout_total
vnode_put_fsm_time_100
vnode_put_fsm_time_95
vnode_put_fsm_time_99
vnode_put_fsm_time_mean
vnode_put_fsm_time_median
vnode_puts
vnode_puts_total
vnode_set_update
vnode_set_update_time_100
vnode_set_update_time_95
vnode_set_update_time_99
vnode_set_update_time_mean
vnode_set_update_time_median
vnode_set_update_total
worker_af1_pool_queuetime_100
worker_af1_pool_queuetime_mean
worker_af1_pool_total
worker_af1_pool_worktime_100
worker_af1_pool_worktime_mean
worker_af2_pool_queuetime_100
worker_af2_pool_queuetime_mean
worker_af2_pool_total
worker_af2_pool_worktime_100
worker_af2_pool_worktime_mean
worker_af3_pool_queuetime_100
worker_af3_pool_queuetime_mean
worker_af3_pool_total
worker_af3_pool_worktime_100
worker_af3_pool_worktime_mean
worker_af4_pool_queuetime_100
worker_af4_pool_queuetime_mean
worker_af4_pool_total
worker_af4_pool_worktime_100
worker_af4_pool_worktime_mean
worker_be_pool_queuetime_100
worker_be_pool_queuetime_mean
worker_be_pool_total
worker_be_pool_worktime_100
worker_be_pool_worktime_mean
worker_node_worker_pool_queuetime_100
worker_node_worker_pool_queuetime_mean
worker_node_worker_pool_total
worker_node_worker_pool_worktime_100
worker_node_worker_pool_worktime_mean
worker_unregistered_queuetime_100
worker_unregistered_queuetime_mean
worker_unregistered_total
worker_unregistered_worktime_100
worker_unregistered_worktime_mean
worker_vnode_pool_queuetime_100
worker_vnode_pool_queuetime_mean
worker_vnode_pool_total
worker_vnode_pool_worktime_100
worker_vnode_pool_worktime_mean
write_once_merge
write_once_put_objsize_100
write_once_put_objsize_95
write_once_put_objsize_99
write_once_put_objsize_mean
write_once_put_objsize_median
write_once_put_time_100
write_once_put_time_95
write_once_put_time_99
write_once_put_time_mean
write_once_put_time_median
write_once_puts
write_once_puts_total
```

Reply:
A comment in the code sharing exactly this would be great. 

```
// Sadly, Riak's documentation on this aspect is dated and doesn't
// include the full lists of available metrics.
// (see https://www.tiot.jp/riak-docs/riak/kv/3.0.4/using/reference/statistics-monitoring/).
// This list of metrics came from running `riak admin status` in our Riak 3.0.10 cluster.
```